File: Roseau-Server/src/main/java/org/alexdev/roseau/game/room/model/Rotation.java
package org.alexdev.roseau.game.room.model;

public class Rotation {
	
	public static byte calculateDirection(int X, int Y, int toX, int toY)
	{
		byte result = 0;
        if (X > toX && Y > toY)
        {
            result = 7;
        }
        else if (X < toX && Y < toY)
        {
            result = 3;
        }
        else if (X > toX && Y < toY)
        {
            result = 5;
        }
        else if (X < toX && Y > toY)
        {
            result = 1;
        }
        else if (X > toX)
        {
            result = 6;
        }
        else if (X < toX)
        {
            result = 2;
        }
        else if (Y < toY)
        {
            result = 4;
        }
        else if (Y > toY)
        {
            result = 0;
        }

        return result;
	}
}

File: src/game/pathfinder/rotation.c
int calculate_human_direction(int x1, int y1, int X2, int Y2) {
    int rotation = 0;

    if (x1 > X2 && y1 > Y2)
        rotation = 7;
    else if (x1 < X2 && y1 < Y2)
        rotation = 3;
    else if (x1 > X2 && y1 < Y2)
        rotation = 5;
    else if (x1 < X2 && y1 > Y2)
        rotation = 1;
    else if (x1 > X2)
        rotation = 6;
    else if (x1 < X2)
        rotation = 2;
    else if (y1 < Y2)
        rotation = 4;
    else if (y1 > Y2)
        rotation = 0;

    return rotation;
}

int calculate_walk_direction(int x, int y, int to_x, int to_y) {
    if (x == to_x) {
        if (y < to_y)
            return 4;
        else
            return 0;
    } else if (x > to_x) {
        if (y == to_y)
            return 6;
        else if (y < to_y)
            return 5;
        else
            return 7;
    } else if (x < to_x) {
        if (y == to_y)
            return 2;
        else if (y < to_y)
            return 3;
        else
            return 1;
    }

    return 0;
}